### PR TITLE
Kmm intreemodules update given CRD changes

### PR DESCRIPTION
**Pull request by cdvultur**
KMM has deprecated the `InTreeModuleToRemove` and moved it to be a list 
- Updated existing WithInTreeModuleToRemove to ensure compatibility on existing tests 
- Added new WithInTreeModulesToRemove
- Pulled new v1beta2 in vendor 

See: https://github.com/rh-ecosystem-edge/kernel-module-management/blob/main/api/v1beta1/module_types.go#L121 for details

### CODE REVIEW

File: pkg/kmm/kernelmapping.go [hunk 1]
```diff
@@ -272,7 +272,28 @@ func (builder *KernelMappingBuilder) WithInTreeModuleToRemove(existingModule str
 		return builder
 	}
 
-	builder.definition.InTreeModuleToRemove = existingModule
+	builder.definition.InTreeModulesToRemove[0] = existingModule
+
+	return builder
+}
+
+// WithInTreeModulesToRemove adds a list of modules to be removed to KernelMapper.
```
[the remark this answers is not in the document]
Comment (cdvultur) [reply pkg/kmm/kernelmapping.go:280]: No. This relates to https://github.com/rh-ecosystem-edge/kernel-module-management/blob/main/api/v1beta1/module_types.go#L130 

```
	// +optional
	// InTreeModulesToRemove specifies any number of  in-tree kernel modules that should be removed (if present)
	// before loading the kernel module from the ContainerImage
	InTreeModulesToRemove []string `json:"inTreeModulesToRemove"`
```

I will leave it as if to stick to the `With<Field>` pattern

**Review by achuzhoy — DISMISSED** [review]

**Comment by kononovn** [comment]
@cdvultur This PR breaks client functionality due to dependency issue.

**Review by kononovn — CHANGES_REQUESTED** [review]
This PR is the root cause of deps conflict .

**Comment by cdvultur** [comment]
closing it for the moment.

**Pull request closed without merging**